src/PointwiseFunctions/AnalyticData/WaveEquation/RegularSphericalWaveKerrSchild.hpp

```diff
@@ -0,0 +1,173 @@
+  };
+  using options =
+      tmpl::list<BlackHoleMass, BlackHoleSpin, BlackHoleCenter, WaveProfile>;
+  static constexpr OptionString help{
+      "A regular spherical wave solution of the covariant wave equation"
+      " with a Kerr black hole (in Kerr-Schild coordinates) as background.\n"
+      "WARNING: This solution should only be used to set initial data."};
+  static std::string name() noexcept { return "RegularSphWaveKerr"; };
+
+  explicit RegularSphericalWaveKerrSchild(
+      double mass, BlackHoleSpin::type dimensionless_spin,
+      BlackHoleCenter::type center, std::unique_ptr<MathFunction<1>> profile,
+      const OptionContext& context = {});
+
+  explicit RegularSphericalWaveKerrSchild(CkMigrateMessage*
+                                          /*unused*/) noexcept {}
+
+  RegularSphericalWaveKerrSchild() = default;
+  RegularSphericalWaveKerrSchild(
+      const RegularSphericalWaveKerrSchild& /*rhs*/) = delete;
+  RegularSphericalWaveKerrSchild& operator=(
+      const RegularSphericalWaveKerrSchild& /*rhs*/) = delete;
+  RegularSphericalWaveKerrSchild(RegularSphericalWaveKerrSchild&&
+                                 /*rhs*/) noexcept = default;
+  RegularSphericalWaveKerrSchild& operator=(
+      RegularSphericalWaveKerrSchild&& /*rhs*/) noexcept = default;
+  ~RegularSphericalWaveKerrSchild() = default;
+
+  // Tags specific to the solution
+  using scalar_wave_tags =
+      tmpl::list<CurvedScalarWave::Pi, CurvedScalarWave::Phi<volume_dim>,
+                 CurvedScalarWave::Psi>;
+  using scalar_wave_dt_tags = db::wrap_tags_in<::Tags::dt, scalar_wave_tags>;
+
+  using DerivLapse = ::Tags::deriv<gr::Tags::Lapse<DataVector>,
+                                   tmpl::size_t<volume_dim>, Frame::Inertial>;
+  using DerivShift =
+      ::Tags::deriv<gr::Tags::Shift<volume_dim, Frame::Inertial, DataVector>,
+                    tmpl::size_t<volume_dim>, Frame::Inertial>;
+  using DerivSpatialMetric = ::Tags::deriv<
+      gr::Tags::SpatialMetric<volume_dim, Frame::Inertial, DataVector>,
+      tmpl::size_t<volume_dim>, Frame::Inertial>;
+  using spacetime_tags = tmpl::list<
+      gr::Tags::Lapse<DataVector>, ::Tags::dt<gr::Tags::Lapse<DataVector>>,
+      DerivLapse, gr::Tags::Shift<volume_dim, Frame::Inertial, DataVector>,
+      ::Tags::dt<gr::Tags::Shift<volume_dim, Frame::Inertial, DataVector>>,
+      DerivShift,
+      gr::Tags::SpatialMetric<volume_dim, Frame::Inertial, DataVector>,
+      ::Tags::dt<
+          gr::Tags::SpatialMetric<volume_dim, Frame::Inertial, DataVector>>,
+      DerivSpatialMetric, gr::Tags::SqrtDetSpatialMetric<DataVector>,
+      gr::Tags::ExtrinsicCurvature<volume_dim, Frame::Inertial, DataVector>,
+      gr::Tags::InverseSpatialMetric<volume_dim, Frame::Inertial, DataVector>>;
+
+  using tags = tmpl::append<spacetime_tags, scalar_wave_tags>;
+
+  // Returns only U
+  tuples::tagged_tuple_from_typelist<scalar_wave_tags> variables(
```

Comment:
Indeed, but I deliberately did this here since this is analytic initial data and not a solution, and it should only be used to set the initial values of `{psi, Phi, Pi}` together.

Can there be a scenario where only a subset of initial variables are needed to be set? E.g. only `psi` or only `Pi` or only the `lapse`? Even if they are needed wont getting them from AnalyticSolutions be a better idea anyway..?